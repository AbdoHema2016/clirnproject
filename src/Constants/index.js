export * from './Paths/LocalPath';
export * from './Paths/Urls';
export * from './appConstants';
