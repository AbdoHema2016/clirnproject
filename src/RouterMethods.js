class RouterMethods {
  constructor() {
    this.props = null;
    this.instance = null;
  }

  setProps(props) {
    this.props = props;
  }

  setInstance(instance) {
    this.instance = instance;
  }
  logout = (token) => {
    const {logout} = this.props;
    logout(token);
  };
}

export const RouterMethodsObj = new RouterMethods();
