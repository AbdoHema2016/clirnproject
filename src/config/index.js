import env from 'react-native-config';

export const IS_PRODUCTION = env.IS_PRODUCTION;
export const API_BASE_URL = env.API_BASE_URL;
export const JUMIO_CALLBACK_URL = env.JUMIO_CALLBACK_URL;
export const APP_NAME = env.APP_NAME;
export const APP_SECRET = env.APP_SECRET;
export const APP_BUNDLE_ID = env.APP_BUNDLE_ID;
export const APK_COMMON_PATH = env.APK_COMMON_PATH;
export const INSTABUG_TOKEN = env.INSTABUG_TOKEN;

export default {
  API_BASE_URL,
  IS_PRODUCTION,
  JUMIO_CALLBACK_URL,
  APP_NAME,
  APP_SECRET,
  APP_BUNDLE_ID,
  APK_COMMON_PATH,
  INSTABUG_TOKEN,
};
