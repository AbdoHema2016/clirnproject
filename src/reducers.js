import {combineReducers} from 'redux';
import {SignInReducer} from './containers/onBoarding/SignIn/redux/reducer';
import {personalDetailsReducer} from './containers/onBoarding/personalDetails/redux/reducer';
import {contactDetailsReducer} from './containers/onBoarding/contactDetails/redux/reducer';
import {otpReducer} from './containers/onBoarding/otp/redux/reducer';
import {feelingReducer} from './containers/health/feeling/redux/reducer';
import {selectTempTypeReducer} from './containers/health/temperature/redux/reducer';
import {healthTestReducer} from './containers/health/healthTest/redux/reducers';
import {userProfileReducer} from './containers/dashBoard/Home/redux/reducer';
import {forgottenPasswordReducer} from './containers/onBoarding/forgotPassword/redux/reducer';
import {settingsReducer} from './containers/dashBoard/Settings/redux/reducer';
import jumio from './containers/onBoarding/verifyDocuments/redux/reducers';
import {notificationsReducer} from './containers/dashBoard/Notifications/redux/reducer';
import {temperatureHistoryReducer} from './containers/dashBoard/Settings/temperatureResults/redux/reducer';
import {changePasswordReducer} from './containers/dashBoard/Settings/passwordSettings/redux/reducer';
import {updateContactDetailsReducer} from './containers/dashBoard/Settings/editPersonalAndContactDetails/redux/reducer';
import {healthTestResultsReducer} from './containers/dashBoard/HealthTests/Results/redux/reducer';
import {biometricReducer} from './containers/Biomitric/redux/reducer';
import {shareHistoryReducer} from './containers/dashBoard/Settings/shareHistory/redux/reducers';
import {companyReducer} from './containers/dashBoard/Company/redux/reducer';
import {vaccinationsReducer} from './containers/dashBoard/vaccinesResults/redux/reducer';
import {LOGOUT_ACTION} from './containers/dashBoard/Settings/redux/types';
import {checkedInReducer} from './containers/dashBoard/CheckedIn/redux/reducer';
import {otherUserProfileReducer} from './containers/dashBoard/OtherUserProfile/redux/reducer';
import {becomeMedicalApproverReducer} from './containers/dashBoard/MedicalApprover/redux/reducer';

const appReducer = combineReducers({
  signIn: SignInReducer,
  forgottenPassword: forgottenPasswordReducer,
  personalDetails: personalDetailsReducer,
  contactDetails: contactDetailsReducer,
  otp: otpReducer,
  feeling: feelingReducer,
  selectTempType: selectTempTypeReducer,
  healthTest: healthTestReducer,
  userProfile: userProfileReducer,
  settingsReducer: settingsReducer,
  jumio,
  notificationsReducer,
  updateContactDetailsReducer,
  changePassword: changePasswordReducer,
  temperatureHistoryReducer,
  healthTestResults: healthTestResultsReducer,
  biometric: biometricReducer,
  shareHistory: shareHistoryReducer,
  companies: companyReducer,
  vaccinations: vaccinationsReducer,
  otherUserProfile: otherUserProfileReducer,
  checkedInReducer,
  becomeMedicalApprover: becomeMedicalApproverReducer,
});
const rootReducer = (state, action) => {
  if (action.type === LOGOUT_ACTION) {
    state = undefined;
  }

  return appReducer(state, action);
};
export default rootReducer;
