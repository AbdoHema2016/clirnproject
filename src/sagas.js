import {all, fork, takeEvery} from 'redux-saga/effects';
import {
  loginWatcher,
  tokenWatcher,
} from './containers/onBoarding/SignIn/redux/saga';
import {personalDetailsWatcher} from './containers/onBoarding/personalDetails/redux/sagas';
import {
  otpWatcher,
  resendotpWatcher,
} from './containers/onBoarding/otp/redux/sagas';
import {
  contactDetailsWatcher,
  getCodeWatcher,
  updatecontactDetailsWatcher,
} from './containers/onBoarding/contactDetails/redux/sagas';
import {healthStatWatcher} from './containers/health/feeling/redux/sagas';
import {temperatureWatcher} from './containers/health/temperature/redux/sagas';
import {
  healthTestWatcher,
  uploadHealthDocs3Watcher,
  updateHealthTestWatcher,
} from './containers/health/healthTest/redux/sagas';
import {
  userProfileWatcher,
  showTemperatureWatcher,
  generateDeepLinkWatcher,
  updateTemperatureWatcher,
  uploadImageS3Watcher,
  showHealthTestInfoWatcher,
  businessRequestWatcher,
  getCompanyDetailsWatcher,
  acceptSharingContactDetailsWatcher,
  meAccessAskShowsWatcher,
  toggleVaccineSharingWatcher,
  generateMeSignWatcher,
} from './containers/dashBoard/Home/redux/sagas';
import {forgotPasswordWatcher} from './containers/onBoarding/forgotPassword/redux/sagas';
import {
  logoutWatcher,
  deleteAccountWatcher,
} from './containers/dashBoard/Settings/redux/sagas';
import {
  getNotoficationsWatcher,
  deleteNotoficationsWatcher,
} from './containers/dashBoard/Notifications/redux/sagas';
import types from './containers/onBoarding/verifyDocuments/redux/actions/types';
import {getJumioCredentialsSaga} from './containers/onBoarding/verifyDocuments/redux/sagas';
import {changePasswordApiWatcher} from './containers/dashBoard/Settings/passwordSettings/redux/sagas';
import {getTempHisotryWatcher} from './containers/dashBoard/Settings/temperatureResults/redux/sagas';
import {
  getHealthTestsWatcher,
  getHealthTestDetailsWatcher,
  deleteHealthTestWatcher,
  uploadRejectedTestImage,
} from './containers/dashBoard/HealthTests/Results/redux/sagas';
import {
  sendPublicKeyWatcher,
  sendSignatureWatcher,
  updatePublicKeyWatcher,
} from './containers/Biomitric/redux/sagas';
import {getcontactDetailsWatcher} from './containers/dashBoard/Settings/editPersonalAndContactDetails/redux/sagas';
import {getShareHistoryWatcher} from './containers/dashBoard/Settings/shareHistory/redux/sagas';
import {
  getCompaniesWatcher,
  showCompanyProfileWatcher,
  setRemoteWorkingWatcher,
  removeCompanyWatcher,
  getRemoteReasonsWatcher,
} from './containers/dashBoard/Company/redux/sagas';
import {
  getVaccinationWatcher,
  deleteVaccineWatcher,
  uploadVaccineImage3Watcher,
  uploadVaccineDetailsWatcher,
  editVaccineDetailsWatcher,
  getVaccineDetailsWatcher,
  verifyVaccineWatcher,
  getVaccineTypesWatcher,
} from './containers/dashBoard/vaccinesResults/redux/sagas';
import {
  getCheckInListWatcher,
  manualCheckoutWatcher,
} from './containers/dashBoard/CheckedIn/redux/saga';
import {approveHealthTestWatcher} from './containers/dashBoard/OtherUserProfile/redux/sagas';
import {becomeMedicalApproverWatcher} from './containers/dashBoard/MedicalApprover/redux/sagas';
export function* rootSaga() {
  yield all([
    fork(loginWatcher),
    fork(tokenWatcher),
    fork(userProfileWatcher),
    fork(updateTemperatureWatcher),
    fork(showTemperatureWatcher),
    fork(forgotPasswordWatcher),
    fork(personalDetailsWatcher),
    fork(contactDetailsWatcher),
    fork(getCodeWatcher),
    fork(otpWatcher),
    fork(resendotpWatcher),
    fork(healthStatWatcher),
    fork(temperatureWatcher),
    fork(healthTestWatcher),
    fork(generateDeepLinkWatcher),
    fork(logoutWatcher),
    fork(uploadImageS3Watcher),
    fork(uploadHealthDocs3Watcher),
    fork(showHealthTestInfoWatcher),
    fork(getNotoficationsWatcher),
    fork(deleteNotoficationsWatcher),
    fork(updatecontactDetailsWatcher),
    fork(changePasswordApiWatcher),
    fork(getTempHisotryWatcher),
    fork(businessRequestWatcher),
    fork(getCompanyDetailsWatcher),
    fork(acceptSharingContactDetailsWatcher),
    fork(getHealthTestsWatcher),
    fork(sendPublicKeyWatcher),
    fork(sendSignatureWatcher),
    fork(updatePublicKeyWatcher),
    fork(getcontactDetailsWatcher),
    fork(updateHealthTestWatcher),
    fork(getShareHistoryWatcher),
    fork(getCompaniesWatcher),
    fork(showCompanyProfileWatcher),
    fork(setRemoteWorkingWatcher),
    fork(removeCompanyWatcher),
    fork(getRemoteReasonsWatcher),
    fork(meAccessAskShowsWatcher),
    fork(deleteAccountWatcher),
    fork(manualCheckoutWatcher),
    fork(getVaccinationWatcher),
    fork(deleteVaccineWatcher),
    fork(uploadVaccineImage3Watcher),
    fork(uploadVaccineDetailsWatcher),
    fork(editVaccineDetailsWatcher),
    fork(toggleVaccineSharingWatcher),
    fork(getCheckInListWatcher),
    fork(getHealthTestDetailsWatcher),
    fork(getVaccineDetailsWatcher),
    fork(deleteHealthTestWatcher),
    fork(uploadRejectedTestImage),
    fork(approveHealthTestWatcher),
    fork(verifyVaccineWatcher),
    fork(becomeMedicalApproverWatcher),
    fork(generateMeSignWatcher),
    fork(getVaccineTypesWatcher),
    yield takeEvery(
      types.GET_JUMIO_CREDENTIALS_REQUESTED,
      getJumioCredentialsSaga,
    ),
  ]);
}
