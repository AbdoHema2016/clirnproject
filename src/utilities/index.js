import * as Constants from '../Constants';
import * as Layout from '../Layout';
import * as HelperFunctions from './HelperFunctions';
import * as OpenExternalURl from './OpenExternalURl';
import * as AndroidDatePicker from './AndroidDatePicker';
import * as Validations from './Validations';
import * as Firebase from './Firebase';
import * as ImagePicker from './ImagePicker';
import * as InitialFunctionCallFromApp from './InitialFunctionCallFromApp';

export * from '../Constants';
export * from '../Layout';
export * from './HelperFunctions';
export * from './OpenExternalURl';
export * from './AndroidDatePicker';
export * from './Validations';
export * from './Firebase';
export * from './ImagePicker';
export * from './InitialFunctionCallFromApp';

export {
  Constants,
  Layout,
  HelperFunctions,
  OpenExternalURl,
  AndroidDatePicker,
  Validations,
  Firebase,
  ImagePicker,
  InitialFunctionCallFromApp,
};
