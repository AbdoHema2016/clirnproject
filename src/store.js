import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import {rootSaga} from './sagas';
const sagaMiddleware = createSagaMiddleware();
const isDevelopmentEnv =
  process && process.env && process.env.NODE_ENV === 'development';
const isProductionEnv =
  process && process.env && process.env.NODE_ENV === 'production';
const logStoreInfo = false;
const middlewares = [sagaMiddleware];
function actionLogger({getState}) {
  return (next) => (action) => {
    let allowed = [
      'jumio',
      'notificationsReducer',
      'otp',
      'companies',
      'signIn',
      'updateContactDetailsReducer',
      'userProfile',
      'vaccinations',
    ];
    let state = getState();
    let loggedStates = Object.keys(state)
      .filter((key) => allowed.includes(key))
      .reduce((obj, key) => {
        return {
          ...obj,
          [key]: state[key],
        };
      }, {});
    console.info({actionType: action.type, loggedStates});
    const returnValue = next(action);
    return returnValue;
  };
}
if (isDevelopmentEnv && logStoreInfo) {
  middlewares.push(createLogger());
}
if (isProductionEnv) {
  middlewares.push(actionLogger);
}

const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddleware.run(rootSaga);
export default store;
