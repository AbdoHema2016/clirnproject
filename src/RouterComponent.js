import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {connect} from 'react-redux';
import {
  StoreToken,
  StoreDeviceToken,
  storeUserIdAction,
} from './containers/onBoarding/SignIn/redux/actions';
import {
  storeAccessTokenAction,
  signUpStepAction,
  storeUserIdInPersonalDetailsAction,
} from './containers/onBoarding/personalDetails/redux/actions';
import {storeStatIdAction} from './containers/health/feeling/redux/actions';
import {profileVisitedBeforeAction} from './containers/dashBoard/Home/redux/actions';
import OnBoardingStack from './Navigation/OnBoardingStack';
import DashBoardStack from './Navigation/DashboardStack';
import AsyncStorage from './utilities/AsyncStorage';
import {Firebase, Constants} from './utilities';
import SplashScreen from 'react-native-splash-screen';
import {navigationRef} from './Navigation/NavigationService';
import ModalsQueue from './services/ModalsQueue';
import {logAnalyticsScreenView} from './utilities/Firebase';
import {RouterMethodsObj} from './RouterMethods';
import {logoutAction} from './containers/dashBoard/Settings/redux/actions';
import {getVaccineTypesAction} from './containers/dashBoard/vaccinesResults/redux/actions';
const {modalIds} = Constants;

class Router extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {};
    this.retrieveData();
    RouterMethodsObj.setProps(this.props);
    RouterMethodsObj.setInstance(this);
  }

  componentDidMount() {
    Firebase.saveTokenFCM((deviceToken) => {
      if (deviceToken != null) {
        this.props.storeDeviceToken(deviceToken);
      }
    });
    setTimeout(() => {
      SplashScreen.hide();
    }, 2000);
  }

  routeNameRef = React.createRef();

  retrieveData = async () => {
    try {
      const token = await AsyncStorage.getItemFromStorage('TOKEN');
      const id = await AsyncStorage.getItemFromStorage('USERID');
      const deviceToken = await AsyncStorage.getItemFromStorage('DEVICE_TOKEN');
      const statId = await AsyncStorage.getItemFromStorage('STATID');
      const accessToken = await AsyncStorage.getItemFromStorage(
        'PERSONALDETAILSACCESSTOKEN',
      );
      const step = await AsyncStorage.getItemFromStorage('STEP');
      const profileVisitedBefore = await AsyncStorage.getItemFromStorage(
        'PROFILEVISITED',
      );
      this.props.storeSignUpStep(step !== null ? step : '0');

      if (profileVisitedBefore === '1') {
        ModalsQueue.hideModal({
          modalId: modalIds.walkThrough,
          hideModalFunction: () => this.props.profileVisitedBefore(true),
        });
      }

      if (profileVisitedBefore !== '1') {
        ModalsQueue.showModal({
          modalId: modalIds.walkThrough,
          showModalFunction: () => this.props.profileVisitedBefore(false),
        });
      }

      if (deviceToken) {
        this.props.storeDeviceToken(deviceToken);
      }
      if (token && id !== null) {
        this.props.storeToken(token, id);
      }
      if (accessToken) {
        this.props.storePersonalDetailsToken(accessToken);
      }
      if (statId !== null) {
        this.props.storeStatId(statId);
      }
      if (id !== null) {
        this.props.storeUserId(id);
        this.props.storeUserIdAction(id);
      }
    } catch (error) {}
  };

  renderNavigator = () => {
    const {token, step} = this.props;
    if (!token && !step) {
      return null;
    }
    if (token) {
      return <DashBoardStack />;
    }
    return <OnBoardingStack />;
  };

  onNavigationStateChange = async (state) => {
    if (!navigationRef?.current?.getCurrentRoute()) {
      return;
    }
    const previousRouteName = this.routeNameRef.current;
    const currentRouteName = navigationRef.current.getCurrentRoute().name;

    if (previousRouteName !== currentRouteName) {
      await logAnalyticsScreenView({currentRouteName});
    }

    // Save the current route name for later comparision
    this.routeNameRef.current = currentRouteName;
  };
  onNavigationStateReady = () => {
    if (!navigationRef?.current?.getCurrentRoute()) {
      return;
    }
    return (this.routeNameRef.current = navigationRef.current.getCurrentRoute().name);
  };

  render() {
    return (
      <NavigationContainer
        ref={navigationRef}
        onReady={this.onNavigationStateReady}
        onStateChange={this.onNavigationStateChange}>
        {this.renderNavigator()}
      </NavigationContainer>
    );
  }
}

const mapStateToProps = ({signIn: {token}, personalDetails: {step}}) => ({
  token,
  step,
});

const mapDispatchToPtops = {
  storeToken: StoreToken,
  profileVisitedBefore: profileVisitedBeforeAction,
  storeDeviceToken: StoreDeviceToken,
  storePersonalDetailsToken: storeAccessTokenAction,
  storeSignUpStep: signUpStepAction,
  storeStatId: storeStatIdAction,
  storeUserId: storeUserIdAction,
  storeUserIdAction: storeUserIdInPersonalDetailsAction,
  logout: logoutAction,
  getVaccineTypes: getVaccineTypesAction,
};

export default connect(mapStateToProps, mapDispatchToPtops)(Router);
