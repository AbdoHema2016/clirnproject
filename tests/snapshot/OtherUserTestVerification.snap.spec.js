import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/store';
import OtherUserProfileData from '../../src/containers/dashBoard/OtherUserProfile/components/profileData';
import {Constants} from '../../src/utilities';
const {screenSource} = Constants;

jest.mock('react-native-gesture-handler', () => {});
jest.mock('@react-navigation/native', () => ({
  withNavigationFocus: (component) => component,
}));

jest.useFakeTimers();

describe('Other user profile screen', () => {
  const VerifiedTestProps = {
    navigation: {
      setOptions: jest.fn(),
    },
    isVerified: true,
    type: screenSource.HEALTH_RESULTS,
    canVerify: true,
    date: '2021-01-25 00:00:00',
  };
  it('Snapshot verified Test', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <OtherUserProfileData {...VerifiedTestProps} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Snapshot not verified Test', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <OtherUserProfileData
            {...{...VerifiedTestProps, isVerified: false}}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Snapshot cannot verify Test', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <OtherUserProfileData {...{...VerifiedTestProps, canVerify: false}} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Snapshot not verified, cannot verify Test', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <OtherUserProfileData
            {...{...VerifiedTestProps, canVerify: false, isVerified: false}}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Snapshot different page source', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <OtherUserProfileData
            {...{...VerifiedTestProps, type: screenSource.EDIT_VACCINE}}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Snapshot verified vaccine', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <OtherUserProfileData
            {...{...VerifiedTestProps, type: screenSource.VACCINE}}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Snapshot not verified vaccine', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <OtherUserProfileData
            {...{
              ...VerifiedTestProps,
              type: screenSource.VACCINE,
              isVerified: false,
            }}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Snapshot cannot verify vaccine', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <OtherUserProfileData
            {...{
              ...VerifiedTestProps,
              type: screenSource.VACCINE,
              canVerify: false,
            }}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Snapshot not verified, cannot verify vaccine', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <OtherUserProfileData
            {...{
              ...VerifiedTestProps,
              type: screenSource.VACCINE,
              canVerify: false,
              isVerified: false,
            }}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
