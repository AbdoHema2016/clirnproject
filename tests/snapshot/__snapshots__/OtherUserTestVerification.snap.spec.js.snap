// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Other user profile screen Snapshot cannot verify Test 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#D1D1D6",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "marginHorizontal": 10,
      "paddingVertical": 13,
    }
  }
>
  <Image
    style={
      Object {
        "borderRadius": 8,
        "height": 24,
        "marginLeft": 8,
        "width": 24,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Futura",
          "fontSize": 18,
        }
      }
    />
    <Text
      accessibilityLabel="verify"
      style={
        Object {
          "color": "#02BF8E",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      [missing "en.healthTest.verified" translation]
    </Text>
    <Text
      style={
        Object {
          "color": "#8D8C91",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      2021-01-25 00:00:00
    </Text>
  </View>
</View>
`;

exports[`Other user profile screen Snapshot cannot verify vaccine 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#D1D1D6",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "marginHorizontal": 10,
      "paddingVertical": 13,
    }
  }
>
  <Image
    style={
      Object {
        "borderRadius": 8,
        "height": 24,
        "marginLeft": 8,
        "width": 24,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Futura",
          "fontSize": 18,
        }
      }
    />
    <Text
      accessibilityLabel="verify"
      style={
        Object {
          "color": "#02BF8E",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      [missing "en.healthTest.verified" translation]
    </Text>
    <Text
      style={
        Object {
          "color": "#8D8C91",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      2021-01-25 00:00:00
    </Text>
  </View>
</View>
`;

exports[`Other user profile screen Snapshot different page source 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#D1D1D6",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "marginHorizontal": 10,
      "paddingVertical": 13,
    }
  }
>
  <Image
    style={
      Object {
        "borderRadius": 8,
        "height": 24,
        "marginLeft": 8,
        "width": 24,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Futura",
          "fontSize": 18,
        }
      }
    />
    <Text
      accessibilityLabel="verify"
      style={
        Object {
          "color": "#02BF8E",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      [missing "en.healthTest.verified" translation]
    </Text>
    <Text
      style={
        Object {
          "color": "#8D8C91",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      2021-01-25 00:00:00
    </Text>
  </View>
</View>
`;

exports[`Other user profile screen Snapshot not verified Test 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#D1D1D6",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "marginHorizontal": 10,
      "paddingVertical": 13,
    }
  }
>
  <Image
    style={
      Object {
        "borderRadius": 8,
        "height": 24,
        "marginLeft": 8,
        "width": 24,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Futura",
          "fontSize": 18,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8D8C91",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      2021-01-25 00:00:00
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#02BF8E",
        "borderRadius": 4,
        "height": 42,
        "right": 8,
        "width": 95,
      }
    }
  >
    <View
      accessibilityLabel="process"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Futura",
            "fontSize": 16,
          }
        }
      >
        [missing "en.STRINGS.PROCESS" translation]
      </Text>
    </View>
  </View>
</View>
`;

exports[`Other user profile screen Snapshot not verified vaccine 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#D1D1D6",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "marginHorizontal": 10,
      "paddingVertical": 13,
    }
  }
>
  <Image
    style={
      Object {
        "borderRadius": 8,
        "height": 24,
        "marginLeft": 8,
        "width": 24,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Futura",
          "fontSize": 18,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8D8C91",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      2021-01-25 00:00:00
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#02BF8E",
        "borderRadius": 4,
        "height": 42,
        "right": 8,
        "width": 95,
      }
    }
  >
    <View
      accessibilityLabel="process"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Futura",
            "fontSize": 16,
          }
        }
      >
        [missing "en.STRINGS.PROCESS" translation]
      </Text>
    </View>
  </View>
</View>
`;

exports[`Other user profile screen Snapshot not verified, cannot verify Test 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#D1D1D6",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "marginHorizontal": 10,
      "paddingVertical": 13,
    }
  }
>
  <Image
    style={
      Object {
        "borderRadius": 8,
        "height": 24,
        "marginLeft": 8,
        "width": 24,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Futura",
          "fontSize": 18,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8D8C91",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      2021-01-25 00:00:00
    </Text>
  </View>
</View>
`;

exports[`Other user profile screen Snapshot not verified, cannot verify vaccine 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#D1D1D6",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "marginHorizontal": 10,
      "paddingVertical": 13,
    }
  }
>
  <Image
    style={
      Object {
        "borderRadius": 8,
        "height": 24,
        "marginLeft": 8,
        "width": 24,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Futura",
          "fontSize": 18,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8D8C91",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      2021-01-25 00:00:00
    </Text>
  </View>
</View>
`;

exports[`Other user profile screen Snapshot verified Test 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#D1D1D6",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "marginHorizontal": 10,
      "paddingVertical": 13,
    }
  }
>
  <Image
    style={
      Object {
        "borderRadius": 8,
        "height": 24,
        "marginLeft": 8,
        "width": 24,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Futura",
          "fontSize": 18,
        }
      }
    />
    <Text
      accessibilityLabel="verify"
      style={
        Object {
          "color": "#02BF8E",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      [missing "en.healthTest.verified" translation]
    </Text>
    <Text
      style={
        Object {
          "color": "#8D8C91",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      2021-01-25 00:00:00
    </Text>
  </View>
</View>
`;

exports[`Other user profile screen Snapshot verified vaccine 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#D1D1D6",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "marginHorizontal": 10,
      "paddingVertical": 13,
    }
  }
>
  <Image
    style={
      Object {
        "borderRadius": 8,
        "height": 24,
        "marginLeft": 8,
        "width": 24,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Futura",
          "fontSize": 18,
        }
      }
    />
    <Text
      accessibilityLabel="verify"
      style={
        Object {
          "color": "#02BF8E",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      [missing "en.healthTest.verified" translation]
    </Text>
    <Text
      style={
        Object {
          "color": "#8D8C91",
          "fontFamily": "Futura",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      2021-01-25 00:00:00
    </Text>
  </View>
</View>
`;
